Sources/MongoSwift/BSON/BSONDecoder.swift

```diff
@@ -228,14 +231,13 @@ internal struct _BSONDecodingStorage {
 
     /// Pops the top container from the stack. 
     fileprivate mutating func popContainer() {
-        precondition(self.containers.count > 0, "Empty container stack.")
+        precondition(!self.containers.isEmpty, "Empty container stack.")
```

Comment:
`empty_count` rule? Neat.

Reply:
yup! this does make me wish `isNotEmpty` or something existed though, as I think it is more readable than `!someCollection.isEmpty`, but it's been pitched various times ([1](https://forums.swift.org/t/isnotempty-on-array/15228), [2](https://forums.swift.org/t/isnonempty/16684)) and shot down by standard library authors. though of course, we could just add it ourselves by extending `Sequence`.